modules/table/src/integrationTest/java/org/apache/ignite/internal/benchmark/CriteriaEmbeddedNodeBenchmark.java

```diff
@@ -0,0 +1,154 @@
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.annotations.Threads;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+
+/**
+ * Benchmark for criteria operation, comparing KV, SQL APIs over embedded node.
+ */
+@State(Scope.Benchmark)
+@Fork(1)
+@Threads(1)
+@Warmup(iterations = 10, time = 2)
+@Measurement(iterations = 20, time = 2)
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.MICROSECONDS)
+@SuppressWarnings({"WeakerAccess", "unused"})
+public class CriteriaEmbeddedNodeBenchmark extends AbstractMultiNodeBenchmark {
+    private static final String SELECT_ALL_FROM_USERTABLE = "select * from usertable where ycsb_key = ?";
+
+    private final Random random = new Random();
+
+    @Param("1")
+    private int clusterSize;
+
+    /**
+     * Benchmark for KV get via embedded node.
+     */
+    @Benchmark
+    public void kvGet(IgniteState state) {
+        state.get(Tuple.create().set("ycsb_key", random.nextInt(DFLT_TABLE_SIZE)));
+    }
+
+    /**
+     * Benchmark for SQL select via embedded node.
+     */
+    @Benchmark
+    public void sqlGet(IgniteState state) {
+        try (var rs = state.sql(SELECT_ALL_FROM_USERTABLE, random.nextInt(DFLT_TABLE_SIZE))) {
+            rs.next();
+        }
+    }
+
+    /**
+     * Benchmark for Criteria get via embedded node.
+     */
+    @Benchmark
+    public void criteriaGet(IgniteState state) {
+        try (Cursor<Tuple> cur = state.query(columnValue("ycsb_key", equalTo(random.nextInt(DFLT_TABLE_SIZE))))) {
+            cur.next();
+        }
+    }
+
+    /**
+     * Benchmark's entry point.
+     */
+    public static void main(String[] args) throws RunnerException {
+        Options opt = new OptionsBuilder()
+                .include(".*" + CriteriaEmbeddedNodeBenchmark.class.getSimpleName() + ".*")
+                .build();
+
+        new Runner(opt).run();
+    }
+
+    /**
+     * Benchmark state for {@link #kvGet(IgniteState)}, {@link #sqlGet(IgniteState)}, {@link #criteriaGet(IgniteState)}.
+     *
+     * <p>Holds {@link Session}.
+     */
+    @State(Scope.Benchmark)
+    public static class IgniteState {
+        private Session session;
+
+        /**
+         * Initializes session and statement.
+         */
+        @Setup
+        public void setUp() {
+            IgniteSql sql = CLUSTER_NODES.get(0).sql();
+
+            session = sql.createSession();
+        }
+
+        /**
+         * Closes resources.
+         */
+        @TearDown
+        public void tearDown() throws Exception {
+            IgniteUtils.closeAll(session);
+        }
+
+        @Nullable Tuple get(Tuple key) {
+            return CLUSTER_NODES.get(0).tables().table(TABLE_NAME).recordView().get(null, key);
```

Comment:
`CLUSTER_NODES.get(0).tables().table(TABLE_NAME).recordView()` this may costs as much as `get()` operation.